gp/cartesian_graph.py

```diff
@@ -1,53 +1,53 @@
 import collections
+import gp
```

Comment:
should not use global import within the package

Reply:
The imports are a bit tricky with mypy and I haven't figured it out yet. Essentially, mypy tries to check all packages that are imported, but some of them don't have proper type annotations yet (e.g. numpy). 

In addition, I haven't figured out how to import the package's own files properly. I agree that we shouldn't have global imports, but with relative imports I am getting errors with mypy all the time. Let's discuss a little bit in person. There is also documentation of this in mypy, but to be honest, I don't understand their documentation or at least my conclusions seem to be wrong. This is their documentation: https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports

Reply:
it's strange because in other parts we do use the relative import of (sub)modules, don't we? does mypy complain about those?

Reply:
Yes, when I use relative imports, mypy throws an error. Using absolute imports solves this, but it's not a good way, so I opened a ticket in their repo to ask about that: https://github.com/python/mypy/issues/8726